### PR TITLE
various minor fixes

### PR DESCRIPTION
**Pull request by oliverkurth**
A few smaller issues.

### CODE REVIEW

File: client/remoterepo.c [hunk 1]
```diff
@@ -668,8 +668,12 @@ TDNFDownloadFile(
...
     {
-        dwError = rename(pszFileTmp, pszFile);
-        BAIL_ON_TDNF_ERROR(dwError);
+        if (rename(pszFileTmp, pszFile) == -1)
+        {
+            dwError = errno;
+            BAIL_ON_TDNF_SYSTEM_ERROR(dwError);
+        }
+
         dwError = chmod(pszFile, S_IRUSR|S_IWUSR|S_IRGRP|S_IROTH);
```
Comment (tapakund) [inline client/remoterepo.c]: we should also do the same for chmod. if we get ret val as -1, look at errno and set dwError accordingly and do BAIL_ON_TDNF_SYSTEM_ERROR

File: tools/cli/main.c [hunk 2]
```diff
@@ -162,6 +162,7 @@ int main(int argc, char **argv)
         {
             TDNFCliShowUsage();
         }
+        dwError = ERROR_TDNF_CLI_NO_SUCH_CMD;
     }
```
Comment (tapakund) [inline tools/cli/main.c:167]: we do not need to BAIL_ON_CLI_ERROR after setting dwError?

Comment (oliverkurth) [reply tools/cli/main.c:167]: Thank you, you are correct. I'll fix it.

File: client/remoterepo.c (hunk 1 above)
Comment (oliverkurth) [reply client/remoterepo.c]: How could I miss that? Thank you! Fixed.

**Review by tapakund — APPROVED** [review]
looks good.

**Pull request merged**